src/com/amazon/ion/impl/IonTypeID.java

```diff
@@ -200,10 +202,9 @@ private IonTypeID(byte id, int minorVersion) {
             byte upperNibble = (byte) ((id >> BITS_PER_NIBBLE) & LOW_NIBBLE_BITMASK);
             // For 0xF0 (delimited end byte) the entire byte is included. This avoids having to create a separate field
             // just to identify this byte.
-            lowerNibble = (id == (byte) 0xF0) ? (byte) 0xF0 : (byte) (id & LOW_NIBBLE_BITMASK);
+            lowerNibble = (id == DELIMITED_END_MARKER) ? DELIMITED_END_MARKER : (byte) (id & LOW_NIBBLE_BITMASK);
```

Comment:
Now that I know what's going on here, I don't see why we need to have this special case. We still have `id` which is a separate field that can be used to identify the `DELIMITED_END_MARKER` byte.

Reply:
That is a good question, and it may be that it's no longer necessary. If you don't mind, I'm going to hold off on changes to this until I resurrect the delimited container support from an old branch I have laying around. This will either go away in that PR, or we can discuss it further there.